lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -277,20 +277,36 @@ Status ProcessElfCore::DoLoadCore() {
 
 void ProcessElfCore::UpdateBuildIdForNTFileEntries() {
   Log *log = GetLog(LLDBLog::Process);
+  m_uuids.clear();
   for (NT_FILE_Entry &entry : m_nt_file_entries) {
     entry.uuid = FindBuidIdInCoreMemory(entry.start);
-    if (log && entry.uuid.IsValid())
-      LLDB_LOGF(log, "%s found UUID @ %16.16" PRIx64 ": %s \"%s\"",
-                __FUNCTION__, entry.start, entry.uuid.GetAsString().c_str(),
-                entry.path.c_str());
+    if (entry.uuid.IsValid()) {
+      m_uuids[entry.path] = entry.uuid;
```

Comment:
This is changing the behavior slightly in that we now take the *last* uuid we find for a file path instead of the first. Hopefully for well-formed these uuids should always match. Probably worth an assert that either its not already in the map or it matches the existing uuid.